src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/FindOperation.java

```diff
@@ -381,34 +386,75 @@ public ReadDocument getNewDocument() {
    *     buildConditions method.
    */
   private List<QueryOuterClass.Query> buildSelectQueries(DBFilterBase.IDFilter additionalIdFilter) {
-    List<List<BuiltCondition>> conditions = buildConditions(additionalIdFilter);
-    if (conditions == null) {
-      return List.of();
-    }
-    List<QueryOuterClass.Query> queries = new ArrayList<>(conditions.size());
-    conditions.forEach(
-        condition -> {
-          if (vector() == null) {
-            queries.add(
-                new QueryBuilder()
-                    .select()
-                    .column(ReadType.DOCUMENT == readType ? documentColumns : documentKeyColumns)
-                    .from(commandContext.namespace(), commandContext.collection())
-                    .where(condition)
-                    .limit(limit)
-                    .build());
-          } else {
-            QueryOuterClass.Query builtQuery = getVectorSearchQuery(condition);
-            final List<QueryOuterClass.Value> valuesList =
-                builtQuery.getValuesOrBuilder().getValuesList();
-            final QueryOuterClass.Values.Builder builder = QueryOuterClass.Values.newBuilder();
-            valuesList.forEach(builder::addValues);
-            builder.addValues(CustomValueSerializers.getVectorValue(vector()));
-            queries.add(QueryOuterClass.Query.newBuilder(builtQuery).setValues(builder).build());
+    AtomicBoolean hasInFilterBesidesIdField = new AtomicBoolean(false);
+    filters.forEach(
+        filter -> {
+          if (filter instanceof DBFilterBase.INFilter) {
+            hasInFilterBesidesIdField.set(true);
+            return; // need to break
           }
         });
 
-    return queries;
+    if (hasInFilterBesidesIdField.get()) {
```

Comment:
sorry, Mahesh, after looking into it for a long time.
I do not think this is doable right now, since resolve() method returns List<DBFilterBase>, I need to iterate through the list.  I can move some codes forward into FindOneAndDeleteCommandResolver, but does not make a big difference.
Do you have some suggestions.